### PR TITLE
Completed bubble_sort and bubble_sort_by

### PR DESCRIPTION
**Pull request by appiersign**
in this pull request we:

- defined the bubble_sort and bubble_sort_by functions
- we added a loop to iterate through the array and compare
- we added checks for nil
- we define and block which handled the sorting for bubble_sort_by
- we refactored the if statements to one liners when necesssary
- we removed print from bubble_sort_by

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  n = array.length - 1
+  n.times do
+    0.upto(n) do |x|
+      if array[x + 1].nil? == false && yield(array[x], array[x + 1]) == 1
+        array[x], array[x + 1] = array[x + 1], array[x]
+      end
+    end
+  end
+  print array
+end
```
Comment (shshamim63) [inline main.rb]: - Bubble sort does not print anything. It just returns a modified arrray.
- While I tried to test your bubble_sort_by with the following code it did not work.
```ruby
bubble_sort_by(%w[hi hello hey]) { |left, right| left.length - right.length }
```
Please resolve the problems.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :space_invader: Required Changes :space_invader: 
___
Hi @appiersign and @aulbytj :wave: 
Awesome job :clap: , Well descriptive readme and your bubble_sort perform properly. But have found some issues that are blocking me to approve this project now. Please see my comments to resolve the problems.
___
### :spades: Mandatory Changes :spades: 
___
1. It seems that your linter is not working properly. Please check this [link](https://github.com/microverseinc/linters-config/tree/master/ruby)
image
2. Your bubble_sort_by is not working properly. Please check the comments below to make it work properly.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Review by pelzolga123 — CHANGES_REQUESTED** [review]
Project requires changes.

**Comment by pelzolga123** [comment]
## Hi Team! 
### You made a good effort, working on changes required previously, one more thing left:
* [x] `bubble_sort_by` has the same issue previous review pointed:

```ruby
 bubble_sort_by(%w[hi hello hey]) { |left, right| left.length - right.length }
``` 
should return `["hi", "hey", "hello"]` as a result. Right now it's returning `["hi", "hello", "hey"]`


### Please kindly fix the above issue and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

**Review by suretrust — APPROVED** [review]
Hi team, 👋 

### Great job implementing the suggested changes. 🙌 Let's get this merged! 👍 

_Good luck with your future projects and happy coding!_ 🎉

**Pull request merged**